source/common/router/rds_impl.cc

```diff
@@ -159,21 +173,121 @@ Router::ConfigConstSharedPtr RdsRouteConfigProviderImpl::config() {
 }
 
 absl::optional<RouteConfigProvider::ConfigInfo> RdsRouteConfigProviderImpl::configInfo() const {
-  if (!subscription_->config_info_) {
+  if (!subscription_->configInfo()) {
     return {};
   } else {
-    return ConfigInfo{subscription_->route_config_proto_,
-                      subscription_->config_info_.value().last_config_version_};
+    return ConfigInfo{subscription_->routeConfiguration(),
+                      subscription_->configInfo().value().last_config_version_};
   }
 }
 
 void RdsRouteConfigProviderImpl::onConfigUpdate() {
   ConfigConstSharedPtr new_config(
-      new ConfigImpl(subscription_->route_config_proto_, factory_context_, false));
+      new ConfigImpl(subscription_->routeConfiguration(), factory_context_, false));
   tls_->runOnAllThreads(
       [this, new_config]() -> void { tls_->getTyped<ThreadLocalConfig>().config_ = new_config; });
 }
 
+VhdsSubscription::VhdsSubscription(
```

Comment:
Should this be split into its own file? I know this entire file has grown to be a monolith, so fine to just punt with a TODO, but I'd like to see it become more decomposed.

Reply:
Looks like it's doable. I'll give it a try.

Reply:
Done. I extended ```RdsConfigProvider``` interface with ```onConfigUpdate``` method (https://github.com/envoyproxy/envoy/pull/6438/files#diff-328729732378ad92588209ae87b94b78R47), but can add another interface just for that, or provide a default implementation right there.